contribs/application/src/test/java/org/matsim/application/options/ShpOptionsTest.java

```diff
@@ -61,4 +63,35 @@ public void all() {
 				.hasSize(Set.copyOf(ft).size());
 
 	}
+
+	@Test
+	public void testGetGeometry() {
+
+		Path input = Path.of(utils.getClassInputDirectory()
+						.replace("ShpOptionsTest", "CreateLandUseShpTest")
+						.replace("options", "prepare"))
+				.resolve("andorra-latest-free.shp.zip");
+
+		Assume.assumeTrue(Files.exists(input));
+
+		ShpOptions shp = new ShpOptions(input, null, null);
+		Geometry geometry1 = shp.getGeometry() ;
+		Geometry geometry2 = null ;
+		Geometry geometry3 = null ;
+
+		List<SimpleFeature> features = shp.readFeatures();
+
+		geometry3 = (Geometry) features.get(0).getDefaultGeometry();
+
+		for(SimpleFeature feature : features) {
+			if(geometry2 == null) {
+				geometry2 = (Geometry) feature.getDefaultGeometry();
+			} else {
+				geometry2 = geometry2.union((Geometry) feature.getDefaultGeometry());
+			}
+		}
+
+		Assert.assertFalse(geometry1.equals(geometry3));
```

Comment:
I figured that it's better to use the geometry's own .equals method as they are explicitely comparing geometries and not just objects. Also, if I use Assert.assertEquals this test will fail. I think this has to do with the way the method .getGeometry and .readFeatures read in the features. Both now produce the same .shp, but if you look closely on their structure, the readFeatures method produces some brackets, which .getGeometry does not produce. Because of this the result of the functions is not EXACTLY the same, which does not matter for the resulting shape, so I think it's fine.

Reply:
Ah, I see. That makes sense.